native/src/main/java/io/ballerina/stdlib/ai/wso2/Native.java

```diff
@@ -58,11 +54,70 @@ private static Object generateJsonSchemaForType(Type t, SchemaGenerationContext
         return switch (impliedType) {
             case JsonType ignored -> generateJsonSchemaForJson();
             case ArrayType arrayType -> generateJsonSchemaForArrayType(arrayType, schemaGenerationContext);
+            case UnionType unionType -> generateUnionTypeSchema(unionType, schemaGenerationContext);
+            case AnnotatableType annotatableType -> getJsonSchemaFromAnnotatableType(annotatableType,
+                    schemaGenerationContext);
+            case MapType mapType  ->  generateMapTypeSchema(mapType, schemaGenerationContext);
             default -> throw ErrorCreator.createError(StringUtils.fromString(
                     "Runtime schema generation is not yet supported for type: " + impliedType.getName()));
         };
     }
 
+    private static Object generateMapTypeSchema(MapType mapType, SchemaGenerationContext schemaGenerationContext) {
+        BMap<BString, Object> schemaMap = createMapValue(TypeCreator.createMapType(PredefinedTypes.TYPE_JSON));
+        schemaMap.put(StringUtils.fromString("type"), StringUtils.fromString("object"));
+//        schemaMap.put(StringUtils.fromString("properties"),createMapValue(TypeCreator.createMapType(PredefinedTypes.TYPE_JSON)));
```

Comment:
Lets remove this and add a comment saying, this should handle in a different structured output method